src/renderer40k.ts

```diff
@@ -722,4 +753,22 @@ function createNotesHead(title: string, notes: BaseNotes[]) {
     return thead;
 }
 
+function addHideAble<T extends Element>(e: T): T {
+    e.classList.add('hide_able')
+    return e;
+}
+
+function toggleHidden(e: Event) {
+    let element = e.target as HTMLElement;
+    if (!element) return;
+    if (element.tagName === 'TD' && element.parentElement) element = element.parentElement as HTMLElement;
```

Comment:
Consider https://developer.mozilla.org/en-US/docs/Web/API/Element/closest to find the closest parent (or self) hideable element: `element?.closest('.hide_able');`. This doesn't have exactly the same semantics as what you have here, and won't work great for nesting, but simplifies things if matches your intent